lib/puppet/functions.rb

```diff
@@ -719,6 +719,16 @@ def cache_param
       inject(:cache)
     end
 
+    # Inject parameter for `Puppet::Pal::CatalogCompiler`
+    def compiler_param
+      inject(:pal_catalog_compiler)
+    end
+
+    # Inject parameter for either `Puppet::Pal::CatalogCompiler` or `Puppet::Pal::ScriptCompiler`
+    def pal_compiler_param
```

Comment:
Shouldn't this be called 'script_compiler_param'? And perhaps the other one should be called 'catalog_compiler_param'? Using 'pal' (or not) to distinguish between them doesn't seem very intuitive.

Reply:
nope, it is for the generic pal compiler; you get either the script compiler or the catalog compiler - then there are two specific ones if you require either script or catalog compiler